shell/platform/android/io/flutter/plugin/editing/TextInputPlugin.java

```diff
@@ -308,6 +313,20 @@ private void setTextInputEditingState(View view, TextInputChannel.TextEditState
         }
     }
 
+    // Samsung's Korean keyboard has a bug where it always attempts to combine characters based on
+    // its internal state, ignoring if and when the cursor is moved programmatically.
+    //
+    // Fully restarting the IMM works around this because it flushes the keyboard's internal state
+    // and stops it from trying to incorrectly combine characters. However this also has some
+    // negative performance implications, so we don't want to apply this workaround in every case.
+    @SuppressLint("NewApi") // New API guard is inline, the linter can't see it.
+    private boolean isRestartAlwaysRequired() {
```

Comment:
Would it be beneficial to cache the results of this? Not sure if and of these checks requires disk IO as you mentioned checking for keyboard may require.

Reply:
Yeah, I think it makes sense. I'm currently caching the result as a final member in the constructor.

Reply:
Just kidding you already do so!